### PR TITLE
Bring up to date (fix compilation with new newlib), remove unused dirs

### PR DESCRIPTION
**Pull request by Aiden-Isik**
See https://github.com/Free60Project/libxenon/pull/34

### CODE REVIEW

File: source/disc.c [hunk 1]
```diff
@@ -81,16 +81,8 @@ const INTERFACE_ID _FAT_disc_interfaces[] = {
...
-}
-
-static const DISC_INTERFACE* get_io_usbstorage_1 (void) {
-        return &__io_usbstorage_1;
-}
-
-static const DISC_INTERFACE* get_io_usbstorage_2 (void) {
-        return &__io_usbstorage_2;
+static const DISC_INTERFACE* get_io_usbstorage (void) {
+	return &__io_usbstorage;
```
Comment (Swizzy) [inline source/disc.c:85]: This looks like a erroneous change - there isn't just 1 USB storage device, it's possible to have 3 of them attached at the same time.

File: source/fatdir.c [hunk 2]
```diff
@@ -588,12 +603,11 @@ int _FAT_dirnext_r (struct _reent *r, DIR_ITER *dirState, char *filename, struct
 	// Make sure there is another file to report on
 	if (! state->validEntry) {
 		_FAT_unlock(&state->partition->lock);
-		r->_errno = ENOENT;
```
Comment (Swizzy) [inline source/fatdir.c:591]: Why was this removed? it looks like we're reducing the details on error?

**Comment by Aiden-Isik** [comment]
Forgot to re-add them after updating the library. Will do that in a bit (although I am not really able to fix LibXenon itself, as it seems like it is very subtly broken after updating the toolchain and I'm not really familiar with, or have the time to look into, it's code and why it's broken. Which makes this PR a bit pointless).
